common/src/auth/ServiceConnectionString.cs

```diff
@@ -0,0 +1,156 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Text;
+
+namespace Microsoft.Azure.Devices.Common.Authorization
+{
+    /// <summary>
+    /// This object handles the connection string for the Azure IoT Services.
+    /// </summary>
+    /// <remarks>
+    /// The connection string contains a set of information that uniquely identify an IoT Service.
+    ///
+    /// A valid connection string shall be in one of the following formats:
+    /// <code>
+    /// HostName=[ServiceName];SharedAccessKeyName=[keyName];SharedAccessKey=[Key]
+    /// HostName=[ServiceName];SharedAccessKeyName=[keyName];SharedAccessSignature=[Signature]
+    /// </code>
+    ///
+    /// This object parse and store the connection string. It is responsible to provide the authorization token too.
+    /// </remarks>
+    public class ServiceConnectionString
+    {
+        protected static readonly TimeSpan DefaultTokenTimeToLive = TimeSpan.FromMinutes(5);
+
+        /// <summary>
+        /// CONSTRUCOR
+        /// </summary>
+        /// <param name="parser">the <see cref="ServiceConnectionStringParser"/> with the connection string content.</param>
+        /// <exception cref="ArgumentNullException">if the provided parser is null.</exception>
+        public ServiceConnectionString(ServiceConnectionStringParser parser)
+        {
+            if (parser == null)
+            {
+                throw new ArgumentNullException(nameof(parser));
+            }
+
+            HostName = parser.HostName;
```

Comment:
One syntactical option to consider is:

HostName = parser?HostName ?? throw new ArgumentNullException(nameof(parser));

You get the same effect, but now we're down to 1 line of code.

It isn't has easy to read, but it is a style that is fairly common, and the case it covers is generally uninteresting code-wise.

Reply:
Do you mean like this? : "HostName = parser.HostName ?? throw new ArgumentNullException(nameof(parser));" 

Is this checking for parser being non null before parse.Hostname?

Reply:
Because you expect parser may be null, you have to add a question mark after it: parser.HostName.

Then if it or HostName is null, the ?? operator will defer to the next part of the statement which is an exception.

Now that I think about it, this would also fire if parser was not null but HostName was. So probably better to leave it the way you had it.

Reply:
Got it. Thanks